### PR TITLE
added "want to sleep a lot" selection.

### PR DESCRIPTION
**Pull request by hasanbal**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -383,13 +383,39 @@ function sortsch(sch){
 	sch.sort(function(a, b){if(b.d < a.d || (b.d == a.d && b.s < a.s)) return 1; return -1;});
 	return sch;
 }
+
+function calc_sleep_score(sch){
+	var sleep_score = 0;
+	for(var i=0; i < sch.length; i++){
+		if(sch[i].p != undefined){
```
Comment (duck2) [inline main.js]: Does this ever happen? Also `null == undefined` in javascript, you might want to use `!==`. (Like in good old php...)

File: main.js [hunk 2]
```diff
@@ -383,13 +383,39 @@ function sortsch(sch){
...
+	for(var i=0; i<Math.min(10,schedules.length); i++){
+		var curMax = {"val":calc_sleep_score(schedules[i]), "index":i};
+
+		for(var j=i+1; j<schedules.length; j++){
+			if(calc_sleep_score(schedules[j]) > curMax.val){
+				curMax.val = calc_sleep_score(schedules[j]);
+				curMax.index = j;
+			}
+		}
+		[schedules[i],schedules[curMax.index]] = [schedules[curMax.index],schedules[i]];
```
Comment (duck2) [inline main.js]: Eh, this swap is [considerably slower](https://jsperf.com/swap-array-vs-variable/42) than using a tmp.

File: main.js [hunk 3]
```diff
@@ -383,13 +383,39 @@ function sortsch(sch){
...
+	for(var i=0; i < sch.length; i++){
+		if(sch[i].p != undefined){
+			sleep_score += sch[i].s;
+		}
+	}
+	return sleep_score;
+}
+
+function sort_for_sleep(){
+	for(var i=0; i<Math.min(10,schedules.length); i++){
```
Comment (duck2) [inline main.js:398]: Is there any way to sort all schedules for this, without dying on 10^6 combinations? maybe using some kind of fancy data structure or something?

**Comment by duck2** [comment]
Left some comments. Mainly I wonder if it's possible to do a scoring scheme for all the schedules without thrashing the browser.

Also configure your git with your github email address...

File: main.js (hunk 1 above)
Comment (hasanbal) [reply main.js]: Actually, this line ignores the Don't Fill blocks. But you are right. I'll change with `!==`.

File: main.js (hunk 2 above)
Comment (hasanbal) [reply main.js]: Okay, I'm fixing with using tmp.

File: main.js (hunk 3 above)
Comment (hasanbal) [reply main.js:398]: Hmm, we can add an attribute something like a `schedules[i].score` and sort for using this value. What do you think about it?

Comment (duck2) [reply main.js:398]: Yeah, probably that would work. Also a lot more elegant than a makeshift limited bubble sort.

**Pull request closed without merging**